### PR TITLE
python3Packages.pyosf: init at 1.0.5

### PR DESCRIPTION
**Pull request by bcdarwin**
###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [NA] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/python-modules/pyosf/default.nix [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+  disabled = isPy27;
+
+  src = fetchFromGitHub {
+    owner = "psychopy";
+    repo = pname;
+    rev = "v${version}";
+    sha256 = "1fkpmylpcbqa9ky111mz4qr1n8pik49gs7pblbb5qx6b54fzl5k2";
+  };
+
+  preBuild = ''export HOME=$TMP'';
```
Comment (IvarWithoutBones) [inline pkgs/development/python-modules/pyosf/default.nix:22]:
```suggestion
  preBuild = "export HOME=$TMP";
```

File: pkgs/development/python-modules/pyosf/default.nix [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+  propagatedBuildInputs = [ requests ];
+
+  doCheck = false;  # requires network access
+  checkInputs = [ pytestCheckHook pytestrunner ];
```
Comment (IvarWithoutBones) [inline pkgs/development/python-modules/pyosf/default.nix:26]: if `doCheck` is set to false, `checkInputs` are useless.

In addition, when no unit tests are available you should use `pythonImportsCheck`.

```suggestion
  doCheck = false; # Tests require network access
  pythonImportsCheck = [ "pyosf" ];
```

**Review by IvarWithoutBones — CHANGES_REQUESTED** [review]
A few suggestions :)

File: pkgs/development/python-modules/pyosf/default.nix (hunk 1 above)
Comment (bcdarwin) [reply pkgs/development/python-modules/pyosf/default.nix]: Changed.

File: pkgs/development/python-modules/pyosf/default.nix (hunk 2 above)
Comment (bcdarwin) [reply pkgs/development/python-modules/pyosf/default.nix]: Done!

**Comment by bcdarwin** [comment]
@IvarWithoutBones - resolved, thanks.

**Comment by bcdarwin** [comment]
@GrahamcOfBorg build python37Packages.pyosf python38Packages.pyosf

**Comment by SuperSandro2000** [comment]
Result of `nixpkgs-review pr 105618` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>3 packages built:</summary>
  <ul>
    <li>python37Packages.pyosf</li>
    <li>python38Packages.pyosf</li>
    <li>python39Packages.pyosf</li>
  </ul>
</details>

File: pkgs/development/python-modules/pyosf/default.nix [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+
+  src = fetchFromGitHub {
+    owner = "psychopy";
+    repo = pname;
+    rev = "v${version}";
+    sha256 = "1fkpmylpcbqa9ky111mz4qr1n8pik49gs7pblbb5qx6b54fzl5k2";
+  };
+
+  preBuild = "export HOME=$TMP";
+  buildInputs = [ pytestrunner ];
```
Comment (IvarWithoutBones) [inline pkgs/development/python-modules/pyosf/default.nix:22]: Tests are disabled, why is this needed?

Comment (bcdarwin) [reply pkgs/development/python-modules/pyosf/default.nix:22]: Because the author has put it in `setup_requires` ...

Comment (bcdarwin) [reply pkgs/development/python-modules/pyosf/default.nix]: I have added a comment to this effect inline.

**Comment by SuperSandro2000** [comment]
Result of `nixpkgs-review pr 105618` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>3 packages built:</summary>
  <ul>
    <li>python37Packages.pyosf</li>
    <li>python38Packages.pyosf</li>
    <li>python39Packages.pyosf</li>
  </ul>
</details>

**Comment by SuperSandro2000** [comment]
Result of `nixpkgs-review pr 105618` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>3 packages built:</summary>
  <ul>
    <li>python37Packages.pyosf</li>
    <li>python38Packages.pyosf</li>
    <li>python39Packages.pyosf</li>
  </ul>
</details>

**Review by IvarWithoutBones — APPROVED** [review]
